src/Codeception/Lib/Driver/MongoDb.php

```diff
@@ -167,6 +169,71 @@ public function load($dumpFile)
         shell_exec($cmd);
     }
 
+    public function loadFromMongoDump($dumpFile)
+    {
+        list($host, $port) = $this->getHostPort();
+        if ($this->user && $this->password) {
+            $cmd = sprintf(
+                'mongorestore --host %s --port %s --username %s --password %s %s',
+                $host,
+                $port,
+                $this->user,
+                $this->password,
+                escapeshellarg($dumpFile)
+            );
+        } else {
+            $cmd = sprintf(
+                'mongorestore --host %s --port %s %s',
+                $host,
+                $port,
+                escapeshellarg($dumpFile)
+            );
+        }
+        shell_exec($cmd);
+    }
+
+    public function loadFromTarGzMongoDump($dumpFile)
+    {
+        list($host, $port) = $this->getHostPort();
+        $getDirCmd = 'tar -tf ' . escapeshellarg($dumpFile)
+            . '| awk \'BEGIN { FS = "/" } ; { print $1 }\' '
+            . '| uniq';
+        $dirCountCmd = $getDirCmd . ' | wc -l';
+        if (trim(shell_exec($dirCountCmd)) !== '1') {
+            throw new ModuleException(
+                $this,
+                'Archive MUST contain single directory with db dump'
+            );
+        }
+        $dirName = trim(shell_exec($getDirCmd));
+        if ($this->user && $this->password) {
+            $cmd = sprintf(
+                'tar -xzf %s '
+                . '&& mongorestore --host %s --port %s --username %s --password %s %s '
+                . '&& rm -r %s',
+                escapeshellarg($dumpFile),
+                $host,
+                $port,
+                $this->user,
+                $this->password,
+                $dirName,
+                $dirName
+            );
+        } else {
+            $cmd = sprintf(
```

Comment:
I think you can avoid using `else` by making this by default value of `$cmd` and appending `--username` and `--password` in case they are set.